File: _experimental/_go/numbers/tests/numbers_iterative_test.go
package tests

import (
	"testing"

	. "example.com/numbers/numbers_iterative"
)

func TestFibonacciIter(t *testing.T) {
	result := Fibonacci(5)
	expected := 5
	if result != expected {
		t.Errorf("Fibonacci(5) = %d; want %d", result, expected)
	} else {
		t.Log("Fibonacci... [OK]")
	}
}

func TestFactorialIter(t *testing.T) {
	result := Factorial(5)
	expected := 120
	if result != expected {
		t.Errorf("Factorial(5) = %d; want %d", result, expected)
	} else {
		t.Log("Factorial... [OK]")
	}
}

func TestSumNumbersIter(t *testing.T) {
	result := SumNumbers(5)
	expected := 15
	if result != expected {
		t.Errorf("SumNumbers(5) = %d; want %d", result, expected)
	} else {
		t.Log("SumNumbers... [OK]")
	}
}

File: _experimental/_rust/numbers/tests/numbers_iterative_test.rs
use numbers::numbers_iterative::*;

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_fibonacci_iterative() {
        assert_eq!(fibonacci(0), 0);
        assert_eq!(fibonacci(1), 1);
        assert_eq!(fibonacci(5), 5);
        assert_eq!(fibonacci(10), 55);
    }

    #[test]
    fn test_factorial_iterative() {
        assert_eq!(factorial(0), 1);
        assert_eq!(factorial(1), 1);
        assert_eq!(factorial(5), 120);
        assert_eq!(factorial(10), 3628800);
    }

    #[test]
    fn test_sum_numbers_iterative() {
        assert_eq!(sum_numbers(0), 0);
        assert_eq!(sum_numbers(1), 1);
        assert_eq!(sum_numbers(5), 15);
        assert_eq!(sum_numbers(10), 55);
    }
}